shell/platform/darwin/macos/framework/Source/FlutterMouseCursorPlugin.mm

```diff
@@ -0,0 +1,156 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#import <objc/message.h>
+
+#import "FlutterMouseCursorPlugin.h"
+#import "flutter/shell/platform/darwin/common/framework/Headers/FlutterCodecs.h"
+
+static NSString* const kMouseCursorChannel = @"flutter/mousecursor";
+
+static NSString* const kActivateSystemCursorMethod = @"activateSystemCursor";
+static NSString* const kKindKey = @"kind";
+
+static NSString* const kKindValueNone = @"none";
+
+/**
+ * Maps a Flutter's constant to a platform's cursor object.
+ *
+ * Returns the arrow cursor for unknown constants, including kSystemShapeNone.
+ */
+static NSCursor* GetCursorForKind(NSString* kind) {
+  // The following mapping must be kept in sync with Flutter framework's
+  // mouse_cursor.dart
+  if ([kind isEqualToString:@"basic"])
+    return [NSCursor arrowCursor];
+  else if ([kind isEqualToString:@"click"])
+    return [NSCursor pointingHandCursor];
+  else if ([kind isEqualToString:@"text"])
+    return [NSCursor IBeamCursor];
+  else if ([kind isEqualToString:@"forbidden"])
+    return [NSCursor operationNotAllowedCursor];
+  else if ([kind isEqualToString:@"grab"])
+    return [NSCursor openHandCursor];
+  else if ([kind isEqualToString:@"grabbing"])
+    return [NSCursor closedHandCursor];
+  else
+    return [NSCursor arrowCursor];
+}
+
+@interface FlutterMouseCursorPlugin ()
+/**
+ * Whether the cursor is currently hidden.
+ */
+@property(nonatomic) BOOL hidden;
+
+/**
+ * Handles the method call that activates a system cursor.
+ *
+ * Returns a FlutterError if the arguments can not be recognized. Otherwise
+ * returns nil.
+ */
+- (FlutterError*)activateSystemCursor:(nonnull NSDictionary*)arguments;
+
+/**
+ * Displays the specified cursor.
+ *
+ * This method unhides the cursor before displaying the cursor, and updates
```

Comment:
Nit:_ remove "This method" here and below; a comment on a method is always about the method, so it's just noise.